python/gtsam/tests/test_Cal3Unified.py

```diff
@@ -14,15 +14,135 @@
+        graph = gtsam.NonlinearFactorGraph()
+        state = gtsam.Values()
+        measured = self.img_point
+        noise_model = gtsam.noiseModel.Isotropic.Sigma(2, 1)
+        pose_key, point_key = P(0), L(0)
+        k = self.stereographic
+        state.insert_pose3(pose_key, gtsam.Pose3())
+        state.insert_point3(point_key, self.obj_point)
+        factor = gtsam.GenericProjectionFactorCal3Unified(measured, noise_model, pose_key, point_key, k)
+        graph.add(factor)
+        score = graph.error(state)
+        self.assertAlmostEqual(score, 0)
+
+    def test_sfm_factor2(self):
+        """Evaluate residual with camera, pose and point as state variables"""
+        r = np.linalg.norm(self.obj_point)
+        graph = gtsam.NonlinearFactorGraph()
+        state = gtsam.Values()
+        measured = self.img_point
+        noise_model = gtsam.noiseModel.Isotropic.Sigma(2, 1)
+        camera_key, pose_key, landmark_key = K(0), P(0), L(0)      
+        k = self.stereographic
+        state.insert_cal3unified(camera_key, k)
+        state.insert_pose3(pose_key, gtsam.Pose3())
+        state.insert_point3(landmark_key, self.obj_point)
+        factor = gtsam.GeneralSFMFactor2Cal3Unified(measured, noise_model, pose_key, landmark_key, camera_key)
+        graph.add(factor)
+        score = graph.error(state)
+        self.assertAlmostEqual(score, 0)
+
+    @unittest.skip("triangulatePoint3 currently seems to require perspective projections.")
+    def test_triangulation(self):
+        """Estimate spatial point from image measurements"""
+        p1 = [-1.0, 0.0, -1.0]
+        p2 = [ 1.0, 0.0, -1.0]
+        q1 = gtsam.Rot3(1.0, 0.0, 0.0, 0.0)
+        q2 = gtsam.Rot3(1.0, 0.0, 0.0, 0.0)
+        obj_point = np.array([0.0, 0.0, 0.0])
+        pose1 = gtsam.Pose3(q1, p1)
+        pose2 = gtsam.Pose3(q2, p2)
+        camera1 = gtsam.PinholeCameraCal3Unified(pose1, self.stereographic)
+        camera2 = gtsam.PinholeCameraCal3Unified(pose2, self.stereographic)
+        cameras = gtsam.CameraSetCal3Unified([camera1, camera2])
+        measurements = gtsam.Point2Vector([k.project(obj_point) for k in cameras])
+        triangulated = gtsam.triangulatePoint3(cameras, measurements, rank_tol=1e-9, optimize=True)
+        self.gtsamAssertEquals(measurements[0], self.img_point)
+        self.gtsamAssertEquals(triangulated, obj_point)
+
+    def test_triangulation_rectify(self):
+        """Estimate spatial point from image measurements using rectification"""
+        p1 = [-1.0, 0.0, -1.0]
+        p2 = [ 1.0, 0.0, -1.0]
+        q1 = gtsam.Rot3(1.0, 0.0, 0.0, 0.0)
+        q2 = gtsam.Rot3(1.0, 0.0, 0.0, 0.0)
+        obj_point = np.array([0.0, 0.0, 0.0])
+        pose1 = gtsam.Pose3(q1, p1)
+        pose2 = gtsam.Pose3(q2, p2)
+        camera1 = gtsam.PinholeCameraCal3Unified(pose1, self.stereographic)
+        camera2 = gtsam.PinholeCameraCal3Unified(pose2, self.stereographic)
```

Comment:
Same deal. If the data is the same for multiple tests, make your life easy by putting it in the `setUp` method.